sdks/go/pkg/beam/core/runtime/harness/datamgr.go

```diff
@@ -29,6 +29,15 @@ import (
 )
 
 const chunkSize = int(4e6) // Bytes to put in a single gRPC message. Max is slightly higher.
+const bufElements = 20     // Number of chunks buffered per reader.
+
+// This is a reduced version of the full gRPC interface to help with testing.
+// TODO(wcn): need a compile-time assertion to make sure this stays synced with what's
```

Comment:
<!--new_thread; commit:c99d4fe17f53af898950a67b6b7d816c570ad9aa; resolved:1-->
If it weren't indirected by the "check in the generated code" step, I'd say that line 107 already does it, since it should be checked by the compiler at that point. Where makeDataChannel is called.